packages/foam-vscode/src/features/preview/wikilink-embed.ts

```diff
@@ -123,4 +118,102 @@ function withLinksRelativeToWorkspaceRoot(
   return text;
 }
 
+interface EmbedNoteExtractor {
+  extract(note: Resource): string;
+}
+
+class FullExtractor implements EmbedNoteExtractor {
+  parser: ResourceParser;
+  workspace: FoamWorkspace;
+
+  constructor(parser: ResourceParser, workspace: FoamWorkspace) {
+    this.parser = parser;
+    this.workspace = workspace;
+  }
+
+  extract(note: Resource) {
+    let noteText = readFileSync(note.uri.toFsPath()).toString();
+    const section = Resource.findSection(note, note.uri.fragment);
+    if (isSome(section)) {
+      const rows = noteText.split('\n');
+      noteText = rows
+        .slice(section.range.start.line, section.range.end.line)
+        .join('\n');
+    }
+    noteText = withLinksRelativeToWorkspaceRoot(
+      noteText,
+      this.parser,
+      this.workspace
+    );
+    return noteText;
+  }
+}
+
```

Comment:
A `class ContentExtractor implements EmbedNoteExtractor...` would go here in the future